### PR TITLE
Fix find -v command on tables (issue #9043)

### PR DESCRIPTION
**Pull request by juanPabloMiceli**
# Description
This PR fixes issue #9043 where find -v was returning empty tables and/or wrong output.
It also refactors some big code chunks with repetitions into it's own functions. 

# User-Facing Changes

# Tests + Formatting
Unit tests added for asserting changes.

# After Submitting

### CODE REVIEW

File: crates/nu-command/tests/commands/find.rs [hunk 1]
```diff
@@ -97,27 +67,96 @@ fn find_with_filepath_search_with_multiple_patterns() {
...
+        ]);
+
+        let actual = nu!(
+            cwd: dirs.test(),
+            "ls | find file --columns [name] --invert | get name | to json -r"
+        );
+
+        assert_eq!(actual.out, r#"["arepas.clu","los.txt","tres.txt"]"#);
+    })
+}
```
Comment (amtoine) [inline crates/nu-command/tests/commands/find.rs:162]: i have trouble wrapping my head around these tests, namely to reproduce them on my machine...

could we simplify them and use `[moe larry bob] | wrap names` as in the tests above?
i have the issue on latest `main` where `[moe larry bob] | wrap names | find m -v` returns a table with emtpy cells.
i think these simpler tests would 
- show the same good behaviour
- be simpler to read and understand

:yum:

Comment (juanPabloMiceli) [reply crates/nu-command/tests/commands/find.rs:162]: Ohh, I didn't know the wrap command! Yes, tests will be way simpler with that

Comment (juanPabloMiceli) [reply crates/nu-command/tests/commands/find.rs:162]: > I have the issue on latest main where [moe larry bob] | wrap names | find m -v returns a table with emtpy cells.
Couldn't reproduce this though. What do you mean by "latest main"?

**Comment by fdncred** [comment]
Seems to be working as advertised.

File: crates/nu-command/tests/commands/find.rs (hunk 1 above)
Comment (amtoine) [reply crates/nu-command/tests/commands/find.rs]: 'twas the latest `git` revision of Nushell at the time :wink:

**Comment by amtoine** [comment]
thanks @juanPabloMiceli for the simpler tests :partying_face:

**Comment by fdncred** [comment]
This PR has broken one of the examples in `find`. @juanPabloMiceli, would you mind looking into it? We cannot do this anymore
image

This issue is documented here as well https://github.com/nushell/nushell/issues/9394

**Comment by juanPabloMiceli** [comment]
Totally!

**Pull request merged**